CenterForOpenScience/osf.io
# [feature] Add bapi latest versioning and latest tag [OSF-8735]

## Purpose
The browsable API (BAPI) returns the earliest (most compatible) version, which can lead to confusion about the current state of the API. This changes the BAPI to use the current version



## Changes
1. Add "version=latest" query and header param (e.g., api.osf.io/v2/?version=latest)
2. Default to latest version for the browsable api (no change for jsonapi or json formats)
3. Add the latest version meta to paginated views
4. Convert tests to Pytest and add more (note: tests were not grouped because there was no time savings, no fixtures, no savings)
5. Remove the fake versions from local-travis.py and local-dist.py which were impacting actual local development. Removed two tests that cannot pass anymore due to using real data.


BEFORE /users/ list view on BAPI would contain:
```
 "meta": {
        "total": 769,
        "per_page": 10,
    }
```
AFTER /users/ list view would contain:
```
 "meta": {
        "total": 769,
        "per_page": 10,
        "version": "2.6"
    }
```

A detail view previously would not have a top level meta, now it does:
<img width="467" alt="screen_shot_2017-11-10_at_10_18_50_am" src="https://user-images.githubusercontent.com/1322421/32664931-b0737d7c-c600-11e7-830e-0d87c7323f63.png">

## Side effects
N/A



## Ticket

https://openscience.atlassian.net/browse/OSF-8735

## QA Notes
1. Confirm that runscope tests are still passing and consider adding one with headers for "latest" version / query param for latest version
2. Confirm that the Browsable api (e.g., staging-api.osf.io/v2/) shows latest version and that adding the latest query param shows the latest version
3. Confirm that normal api requests don't default to latest versionn
